src/artifactory_auth_store.rs

```diff
@@ -0,0 +1,72 @@
+use crate::error::ForemanError;
+use crate::{error::ForemanResult, fs};
+use artiaa_auth::{error::ArtifactoryAuthError, Credentials};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use std::{
+    ops::{Deref, DerefMut},
+    path::Path,
+};
+/// Contains stored user tokens that Foreman can use to download tools.
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct ArtifactoryAuthStore {
+    tokens: HashMap<String, Credentials>,
+}
+
+impl Deref for ArtifactoryAuthStore {
+    type Target = HashMap<String, Credentials>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.tokens
+    }
+}
+
+impl DerefMut for ArtifactoryAuthStore {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.tokens
+    }
+}
+
+impl ArtifactoryAuthStore {
+    pub fn set_token(auth_file: &Path, key: &str, token: &str) -> ForemanResult<()> {
+        let contents = fs::try_read_to_string(auth_file)?;
+
+        let mut store: ArtifactoryAuthStore = if let Some(contents) = contents {
+            serde_json::from_str(&contents).map_err(|err: serde_json::Error| {
+                ForemanError::ArtiAAError {
+                    error: ArtifactoryAuthError::auth_parsing(auth_file, err.to_string()),
+                }
+            })?
+        } else {
+            ArtifactoryAuthStore::default()
+        };
+
+        store.insert(
+            key.to_owned(),
+            Credentials {
+                username: "".to_owned(),
```

Comment:
The artifactory auth format assumes a username field exists, but it isn't actually used to authenticate to Artifactory